### DISCUSSION TITLE
Demo on the M5

### DISCUSSION THREAD

**Discussion by CyberSmash** [PlummersSoftwareLLC/NightDriverStrip#42] [Q&A]
Does the demo build and run on the M5StickCPlus? The code builds, and uploads but doesn't do anything, including show anything on the screen. No amount of pressing buttons matters either.

My current setup is the following:

Environment (modified by me):
```
[env:demo]
platform      = espressif32
board         = m5stick-c
monitor_speed = 115200
upload_speed  = 1500000
upload_port   = COM5                    ; This will vary by machine
build_flags   = -DDEMO=1
                -DM5STICKCPLUS=1
                -std=gnu++17
                -Ofast 
lib_deps      = ${m5stick-c-plus.lib_deps}
```

globals.h
```
 #define ENABLE_WIFI             0   // Connect to WiFi
    #define INCOMING_WIFI_ENABLED   0   // Accepting incoming color data and commands
    #define TIME_BEFORE_LOCAL       2   // How many seconds before the lamp times out and shows local contexnt
    #define ENABLE_NTP              0   // Set the clock from the web
    #define ENABLE_OTA              0   // Accept over the air flash updates
    #define USE_OLED                1   // Found on another discussion
    #define USE_TFT                 0   // Set to 1 if you have the Heltec module w/TFT 
```

I'm building and running with `platformio run -t upload -e demo`

Thanks in advance

**Comment by davepl** [comment]
Demo only controls LEDs.  It doesn’t put anything on the display or respond to any buttons.

**Reply by CyberSmash** [reply]
Thanks for the prompt reply Dave. I'm really excited about this project.

> Demo only controls LEDs.  It doesn’t put anything on the display or respond to any buttons.

Good, that was my original assumption. But I had found this discussion [(23)](https://github.com/PlummersSoftwareLLC/NightDriverStrip/discussions/23) and wasn't sure. 

I've recompiled everything starting from the base project, my globals.h looks like this: 

```c
#define MATRIX_WIDTH            144
#define MATRIX_HEIGHT           1
#define NUM_LEDS                (MATRIX_WIDTH*MATRIX_HEIGHT)
#define NUM_CHANNELS            1
#define LED_PIN0                26
#define NUM_RINGS               5
#define RING_SIZE_0             24

    #define POWER_LIMIT_MW       5000   // 1 amp supply at 5 volts assumed
// Once you have a working project, selectively enable various additional features by setting
// them to 1 in the list below.  This DEMO config assumes no audio (mic), or screen, etc.

#define ENABLE_WIFI             0   // Connect to WiFi
#define INCOMING_WIFI_ENABLED   0   // Accepting incoming color data and commands
#define TIME_BEFORE_LOCAL       2   // How many seconds before the lamp times out and shows local contexnt
#define ENABLE_NTP              0   // Set the clock from the web
#define ENABLE_OTA              0   // Accept over the air flash updates
#define USE_TFT                 0   // Set to 1 if you have the Heltec module w/TFT 
```

And I've reset my platform.ini file to the following (unchanged / irrelevant sections excluded):
```ini
[env:demo]
platform      = espressif32
board         = esp32dev
monitor_speed = 115200
upload_speed  = 1500000
upload_port   = COM14                    ; This will vary by machine
build_flags   = -DDEMO=1
                -std=gnu++17
                -Ofast 
```

I've also tried changing the `board` to `m5stick-c`, adding `-DM5STICKPLUS=1` and adding the `lib_deps` for the `m5stick-c-plus.lib_deps` to the demo environment, with no change. 

I took the oscilloscope to pin 26 on the board and there isn't any signal. I've also hooked up an external power supply providing 5V to the LED strip to power it externally being sure to ground everything to the same common ground. Still nothing (which is expected considering there's no signal on my signal pin).

For S's and G's I decided to switch the pin to G0 (as opposed to G26), as the scope was telling me that that pin was always pulled high. This resulted in no real change as the pin is still just pulled high.

I have a heltec board coming in on Thursday and can switch out to that, but I imagined the M5 shouldn't have any trouble, and didn't know if there was something silly I needed to do to get the M5 board to work with the demo.

**Comment by davepl** [comment]
If you’re using an M5StickCPlus, I’d recommend just building one of the projects that is already set up for it, like SPECTRUM.  Then you can prune out what you don’t need, perhaps, rather than trying to add the right ingredients one by one?

If you build and upload spectrum, you should get some stuff on the display and it will be outputting an LED signal on PIN26.  

If I get a chance soon I’ll try to get DEMO working on the M5 out of the box, as it probably should :-). I’d appreciate it if you could open an issue about this and assign it to me.  “DEMO doesn’t work on M5StickCPlus” would be a good title!

- Dave

**Reply by CyberSmash** [reply]
> I’d appreciate it if you could open an issue about this and assign it to me.

See issue #43 . I don't appear to have the ability to assign the ticket. Let me know if it needs more details.

> If you’re using an M5StickCPlus, I’d recommend just building one of the projects that is already set up for it, like SPECTRUM.  
Then you can prune out what you don’t need, perhaps, rather than trying to add the right ingredients one by one?

Yeah I completely agree. That's what I tried first, and then after reverting everything, going back to the demo, and submitting this discussion tried again after you suggested it, this time with a completely clean environment. Sadly, spectrum doesn't work and does the  same thing as the demo script. Just to check to make sure that my stick wasn't completely borked, I uploaded a simple "hello world" demo application to the m5stickCPlus and that "hello world" script worked just fine.

**Comment by robertlipe** [comment]
https://github.com/PlummersSoftwareLLC/NightDriverStrip/issues/43 has been accepted and closed. Should this related conversation also be marked as answered, @CyberSmash ?
